src/utils/types.rs

```diff
@@ -13,3 +13,25 @@ pub(crate) fn truncate(data: &[u8]) -> Hash {
     hash[..n].copy_from_slice(&data[..n]);
     hash
 }
+
+#[derive(Debug, Clone)]
+/// Type of dataset's elements with an optional index
+pub struct Element<E> {
+    /// Set element data
+    pub data: E,
+    /// ID of the element
+    pub index: Option<u64>,
+}
+
```

Comment:
We could add a function to compare two elements to simplify the sorting in */proof.rs